hal/lp_ticker_api.h

```diff
@@ -49,37 +49,126 @@ void lp_ticker_irq_handler(void);
 
 /** Initialize the low power ticker
  *
+ * Pseudo Code:
+ * @code
+ * void lp_ticker_init()
+ * {
+ *     // Enable clock gate so processor can read LPTMR registers
+ *     POWER_CTRL |= POWER_CTRL_LPTMR_Msk;
+ *
+ *     // Disable the timer and ensure it is powered down
+ *     LPTMR_CTRL &= ~(LPTMR_CTRL_ENABLE_Msk | LPTMR_CTRL_COMPARE_ENABLE_Msk);
+ *
+ *     // Configure divisors - no division necessary
+ *     LPTMR_PRESCALE = 0;
+ *     LPTMR_CTRL |= LPTMR_CTRL_ENABLE_Msk;
+ *
+ *     // Install the interrupt handler
+ *     NVIC_SetVector(LPTMR_IRQn, (uint32_t)lp_ticker_irq_handler);
+ *     NVIC_EnableIRQ(LPTMR_IRQn);
+ * }
+ * @endcode
  */
 void lp_ticker_init(void);
 
 /** Read the current counter
  *
- * @return The current timer's counter value in microseconds
+ * @return The current timer's counter value in ticks
```

Comment:
Would it be possible to specify what a tick is and the relationship to the value return and the `ticker_info_t` returned by `lp_ticker_get_info`.
Could you also detail the expected behavior if this function is called while `lp_ticker_init` hasn't been called.

Reply:
Added information on how ticker_read relates to ticker_get_info. In the Undefined behavior section I have that calling any function other ticker function before ticker_init is undefined behavior. If you want I could add that info to each of the functions.